curriculum/challenges/english/25-front-end-development/lab-password-generator/66f53dc2c5bd6a11d6c3282f.md

```diff
@@ -0,0 +1,135 @@
+5. You should have a `console.log` that logs the message `"Generated password:"` followed by the `password` variable. 
+
+# --hints--
+
+You should have a `generatePassword` function with a parameter. You can name the parameter whatever you like.
+
+```js
+assert.isFunction(generatePassword);
+assert.isTrue(generatePassword.length === 1);
+```
+
+Your `generatePassword` function should return a string.
+
+```js
+const result = generatePassword(5);
+assert.isString(result);
+```
+
+Your `generatePassword` function should return a new string that is the correct length.
+
+```js
+const length = 8;
+const password = generatePassword(length);
+assert.strictEqual(password.length, length);
+```
+
+Your function should return a randomly generated password with valid characters.
+
+```js
+const validChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()";
+const password = generatePassword(10);
+for (let char of password) {
+    assert.include(validChars, char);
+}
+```
+
+Your function should return a new random string each time it is called.
+
+```js
+const password1 = generatePassword(10);
+const password2 = generatePassword(10);
+assert.notStrictEqual(password1, password2);
+```
+
+You should have a variable called `strLen`.
+
+```js
+assert.match(code, /\bconst\s+strLen\b/);
+```
+
+You should prompt the user for the password length.
+
+```js
+const originalPrompt = window.prompt;
+let promptCalled = false;
+window.prompt = () => {
+    promptCalled = true;
+    return '10'; 
+};
+eval(code);
+assert.isTrue(promptCalled);
+window.prompt = originalPrompt;
+```
+
+Your `strLen` variable should be a number.
+
+```js
+assert.match(code, /\bconst\s+strLen\s*=\s*parseInt\s*\(\s*prompt\s*\(\s*["'`].*?["'`]\s*\)\s*,\s*\d+\s*\)/,);
+```
+
+You should have a `password` variable.
+
+```js
+assert.match(code, /\bconst\s+password\b/);
+```
+
+Your `password` variable should be a string.
+
+```js
+const passwordType = typeof password;
+assert.strictEqual(passwordType, 'string');
+```
+
+You should log the generated password to the console.
+
+```js
+const condition1 = /console\.log\(\s*["']Generated\s+password:\s*["']\s*\+\s*password\s*\);?/gm.test(code);
+const condition2 = /console\.log\(\s*`Generated\s+password:\s*\$\{password\}`\s*\);?/gm.test(code);
+assert.isTrue(condition1 || condition2);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```js
+
+```
+
+# --solutions--
+
+```js
+const generatePassword = (length) => {
+    const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()";
+    let password = "";
+    for (let i = 0; i < length; i++) {
+        const randomIndex = Math.floor(Math.random() * chars.length);
+        password += chars[randomIndex];
+    }
+    return password;
+};
+
+const strLen = parseInt(prompt("Enter the desired length for the password:"), 10);
```

Comment:
> [The tests are failing on this PR](https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/12106469459/job/33772579600?pr=56330) @ihechikara - want to take a look at those? Seems like one or more of the tests are flaky or timing out or something. Let us know if you need a hand.

The first test is failing as it is waiting for an input via an alert. When there is no input, the test times out.

Reply:
I think we should remove this part and simply ask to call `generatePassword` with a number. If prompt is called by the `generatePassword` function we can override it but in this way and without using `--before-user-code--` I don't see how to make it work.